File: homework01/vigener.go
package vigenere


func EncryptVigenere(plaintext string, keyword string) string {
	var ciphertext string

	var i,j int
	i=0
	for j=0; j<len(plaintext);j++{
		let:=int(plaintext[j])
		comp:=int(plaintext[j])
		if i>=len(keyword){
			i=0
		}
		if (int('A') <= let) && (let <= int('Z')) || ((int('a') <= let) && (let <= int('z'))) {
			if (int('A') <= int(keyword[i])) && (int(keyword[i]) <= int('Z')) {
				let = let + int(keyword[i]) - int('A')
			}
			if (int('a') <= int(keyword[i])) && (int(keyword[i]) <= int('z')) {
				let = let + int(keyword[i]) - int('a')
			}
		}
			if (int('A') <= comp) && (comp <= int('Z') && (int('Z') < let)) || (int('a') <= comp) && (comp <= int('z') && (int('z') < let)){
				let-=26
			}
		i+=1
		ciphertext+=string(let)
	}

	return ciphertext
}

func DecryptVigenere(ciphertext string, keyword string) string {
	var plaintext string

	var i,j int
	i=0
	for j=0; j<len(ciphertext);j++ {
		let := int(ciphertext[j])
		comp := int(ciphertext[j])
		if (int('A') <= let) && (let <= int('Z')) || (int('a') <= let && let <= int('z')) {
			if i >= len(keyword) {
				i = 0
			}
			if (int('A') <= int(keyword[i])) && (int(keyword[i]) <= int('Z')) {
				let = let - int(keyword[i]) + int('A')
			}
			if (int('a') <= int(keyword[i])) && (int(keyword[i]) <= int('z')) {
				let = let - int(keyword[i]) + int('a')
			}
			if (int('A') <= comp) && (comp <= int('Z') && (int('A') > let)) || (int('a') <= comp) && (comp <= int('z') && (int('a') > let)) {
				let += 26
			}
		}
		i += 1

		plaintext += string(let)
	}
	return plaintext
}

File: homework01/vigener.py
def encrypt_vigenere(plaintext, keyword):
    """
    >>> encrypt_vigenere("PYTHON", "A")
    'PYTHON'
    >>> encrypt_vigenere("python", "a")
    'python'
    >>> encrypt_vigenere("ATTACKATDAWN", "LEMON")
    'LXFOPVEFRNHR'
    """
    i = 0
    ciphertext = ''
    for lit in plaintext:
        if 'A' <= lit <= 'Z' or 'a' <= lit <= 'z':
            if i > len(keyword) - 1:
                i = 0
            if 'A' <= keyword[i] <= 'Z':
                lit1 = ord(lit) + (ord(keyword[i]) - 65)
            elif 'a' <= keyword[i] <= 'z':
                lit1 = ord(lit) + (ord(keyword[i]) - 97)
            if 'A' <= lit <= 'Z' and lit1 > ord('Z') or \
                    'a' <= lit <= 'z' and lit1 > ord('z'):
                lit1 -= 26
        i += 1
        ciphertext += chr(lit1)
    return ciphertext


def decrypt_vigenere(ciphertext, keyword):
    """
    >>> decrypt_vigenere("PYTHON", "A")
    'PYTHON'
    >>> decrypt_vigenere("python", "a")
    'python'
    >>> decrypt_vigenere("LXFOPVEFRNHR", "LEMON")
    'ATTACKATDAWN'
    """
    i = 0
    plaintext = ''
    for lit in ciphertext:
        if 'A' <= lit <= 'Z' or 'a' <= lit <= 'z':
            if i > len(keyword) - 1:
                i = 0
            if 'A' <= keyword[i] <= 'Z':
                lit1 = ord(lit) - (ord(keyword[i]) - 65)
            elif 'a' <= keyword[i] <= 'z':
                lit1 = ord(lit) - (ord(keyword[i]) - 97)
            if 'A' <= lit <= 'Z' and lit1 < ord('A') or \
                    'a' <= lit <= 'z' and lit1 < ord('a'):
                lit1 += 26
        i += 1
        plaintext += chr(lit1)
    return plaintext
